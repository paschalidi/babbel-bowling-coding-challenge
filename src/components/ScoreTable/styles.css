table {
    border-spacing: 0 !important;
    margin: 0 auto;
}

th {
    text-align: center;
    min-width: 30px;
    min-height: 30px;
    height: 30px;
}

th.border {
    border: dotted white;
    border-top-width: 0px;
    border-right-width: 0px;
    border-left-width: 2px;
    border-bottom-width: 2px;
}

.border {
    border: 1px solid white;

}
