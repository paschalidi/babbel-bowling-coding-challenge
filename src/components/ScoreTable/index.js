/**
 *
 * ScoreTable
 *
 */
/* eslint-disable import/first */
/* eslint-disable react/jsx-filename-extension */
/* eslint-disable react/prop-types */

import React, { Fragment } from 'react';
import PT from 'prop-types';
import { Grid } from 'semantic-ui-react';
import { LAST_ROUND } from '../../utils/constants';
// eslint-disable-next-line
import styles from './styles.css';


const ScoreTable = ({ indexOfActivePlayer, players }) =>
  <Grid.Row columns={11}>
    {
      Object.keys(players[indexOfActivePlayer].scores).map(index => {
          const score = players[indexOfActivePlayer].scores[index];
          return (
            <Grid.Column key={index}>
              <table className='border'>
                <tbody>
                {
                  parseInt(index, 10) !== LAST_ROUND
                    ?
                    <Fragment>
                      <tr>
                        <th>{score[0]}</th>
                        <th className='border'>{score[1]}</th>
                      </tr>
                      <tr>
                        <th />
                        <th>{score[0] + score[1]}</th>
                      </tr>
                    </Fragment>
                    :
                    <Fragment>
                      <tr>
                        <th>{score[0]}</th>
                        <th className='border'>{score[1]}</th>
                        <th className='border'>{score[2]}</th>
                      </tr>
                      <tr>
                        <th />
                        <th>{score[0] + score[1] + score[2]}</th>
                      </tr>
                    </Fragment>
                }
                </tbody>
              </table>
            </Grid.Column>);
        }
      )
    }
  </Grid.Row>;

ScoreTable.defaultProps = {
  indexOfActivePlayer: 0,
  players: {
    0: {
      id: 0, name: `Player 1`, bonusRounds: 0, totalScore: 0, scores: {
        0: { 0: null, 1: null },
        1: { 0: null, 1: null },
        2: { 0: null, 1: null },
        3: { 0: null, 1: null },
        4: { 0: null, 1: null },
        5: { 0: null, 1: null },
        6: { 0: null, 1: null },
        7: { 0: null, 1: null },
        8: { 0: null, 1: null },
        9: { 0: null, 1: null, 2: null }
      }
    }
  }
};
ScoreTable.ScoreTable = {
  indexOfActivePlayer: PT.number.isRequired,
  players: PT.object.isRequired
};

export default ScoreTable;
