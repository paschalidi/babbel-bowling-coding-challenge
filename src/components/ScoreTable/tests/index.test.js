import React from 'react';
import renderer from 'react-test-renderer';
import ScoreTable from '../index';


test('ScoreTable renders', () => {
  const component = renderer.create(
    <ScoreTable />
  );
  let tree;

  tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
