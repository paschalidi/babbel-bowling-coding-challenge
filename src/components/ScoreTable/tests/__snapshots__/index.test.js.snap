// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ScoreTable renders 1`] = `
<div
  className="eleven column row"
>
  <div
    className="column"
  >
    <table
      className="border"
    >
      <tbody>
        <tr>
          <th />
          <th
            className="border"
          />
        </tr>
        <tr>
          <th />
          <th>
            0
          </th>
        </tr>
      </tbody>
    </table>
  </div>
  <div
    className="column"
  >
    <table
      className="border"
    >
      <tbody>
        <tr>
          <th />
          <th
            className="border"
          />
        </tr>
        <tr>
          <th />
          <th>
            0
          </th>
        </tr>
      </tbody>
    </table>
  </div>
  <div
    className="column"
  >
    <table
      className="border"
    >
      <tbody>
        <tr>
          <th />
          <th
            className="border"
          />
        </tr>
        <tr>
          <th />
          <th>
            0
          </th>
        </tr>
      </tbody>
    </table>
  </div>
  <div
    className="column"
  >
    <table
      className="border"
    >
      <tbody>
        <tr>
          <th />
          <th
            className="border"
          />
        </tr>
        <tr>
          <th />
          <th>
            0
          </th>
        </tr>
      </tbody>
    </table>
  </div>
  <div
    className="column"
  >
    <table
      className="border"
    >
      <tbody>
        <tr>
          <th />
          <th
            className="border"
          />
        </tr>
        <tr>
          <th />
          <th>
            0
          </th>
        </tr>
      </tbody>
    </table>
  </div>
  <div
    className="column"
  >
    <table
      className="border"
    >
      <tbody>
        <tr>
          <th />
          <th
            className="border"
          />
        </tr>
        <tr>
          <th />
          <th>
            0
          </th>
        </tr>
      </tbody>
    </table>
  </div>
  <div
    className="column"
  >
    <table
      className="border"
    >
      <tbody>
        <tr>
          <th />
          <th
            className="border"
          />
        </tr>
        <tr>
          <th />
          <th>
            0
          </th>
        </tr>
      </tbody>
    </table>
  </div>
  <div
    className="column"
  >
    <table
      className="border"
    >
      <tbody>
        <tr>
          <th />
          <th
            className="border"
          />
        </tr>
        <tr>
          <th />
          <th>
            0
          </th>
        </tr>
      </tbody>
    </table>
  </div>
  <div
    className="column"
  >
    <table
      className="border"
    >
      <tbody>
        <tr>
          <th />
          <th
            className="border"
          />
        </tr>
        <tr>
          <th />
          <th>
            0
          </th>
        </tr>
      </tbody>
    </table>
  </div>
  <div
    className="column"
  >
    <table
      className="border"
    >
      <tbody>
        <tr>
          <th />
          <th
            className="border"
          />
          <th
            className="border"
          />
        </tr>
        <tr>
          <th />
          <th>
            0
          </th>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;
